### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,12 +25,16 @@ const { wrap } = require('lambda-cfn-hook');
...
-});
+}, {/* options */});
 
 ```
 
 and attach the `hook` function to a [custom resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html) as a Lambda function in CloudFormation.
 
+## Options
+
+* `silent` _boolean_: Default `false`. If `true`, then callback failure does not throw error
```
Comment (MacMcIrish) [inline README.md]:
```suggestion
* `silent` _boolean_: Default `false`. If `true`, then response not sent for invalid events
```
Maybe something more specific? The function can still throw if the `submit` fails on the request

Comment (simlu) [reply README.md]:
```suggestion
* `silent` _boolean_: Default `false`. If set to `true`, then no error is thrown if an event is received that isn't originating from a custom resource life-cycle lambda hook. Useful if the lambda handler is multi purpose.
```

File: src/logic/wrap.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
+const get = require('lodash.get');
 const request = require('request-promise');
+const Joi = require('joi-strict');
 const { wrap } = require('lambda-async');
 const { logger, abbrev } = require('lambda-monitor-logger');
 
-const submit = async (event, context, success) => {
+const submit = async (event, context, opts) => {
```
Comment (simlu) [inline src/logic/wrap.js]:
```suggestion
const submit = async ({ event, context, success, silent }) => {
```

File: test/logic/wrap.spec.js [hunk 3]
```diff
@@ -5,25 +5,26 @@ const wrap = require('../../src/logic/wrap');
...
   beforeEach(({ fixture }) => {
-    executor = async (
-      fn,
+    executor = async ({
+      fn = () => 'ok',
       event = fixture('event'),
-      context = fixture('context')
-    ) => new Promise((resolve) => wrap(fn)(
+      context = fixture('context'),
+      opts
```
Comment (simlu) [inline test/logic/wrap.spec.js]:
```suggestion
      opts = undefined
```

**Pull request merged**